src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1664,20 +1664,18 @@ void TorrentHandle::handleSaveResumeDataFailedAlert(const libtorrent::save_resum
 
 void TorrentHandle::handleFastResumeRejectedAlert(const libtorrent::fastresume_rejected_alert *p)
 {
-    qDebug("/!\\ Fast resume failed for %s, reason: %s", qUtf8Printable(name()), p->message().c_str());
-
-    updateStatus();
     if (p->error.value() == libt::errors::mismatching_file_size) {
         // Mismatching file size (files were probably moved)
-        LogMsg(tr("File sizes mismatch for torrent '%1', pausing it.").arg(name()), Log::CRITICAL);
+        pause();
         m_hasMissingFiles = true;
-        if (!isPaused())
-            pause();
+        LogMsg(tr("File sizes mismatch for torrent '%1', pausing it.").arg(name()), Log::CRITICAL);
     }
     else {
         LogMsg(tr("Fast resume data was rejected for torrent '%1'. Reason: %2. Checking again...")
             .arg(name(), QString::fromStdString(p->message())), Log::CRITICAL);
     }
+
+    updateStatus();
```

Comment:
This operation is costly, AFAIK. We needed it before to check correctly for "paused" state, IIRC. Is it still needed here?

Reply:
>This operation is costly, AFAIK.

Yes, that's why I moved it to the bottom.

>Is it still needed here?

I think it is needed to have the correct update of the status for GUI/WebUI). updateStatus() calls updateState() too

Reply:
GUI/WebUI can get updated status a little later... Otherwise your solution can help in case of single corrupted torrent only. All remaining will have the same delay as before.